src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -70,17 +70,17 @@ internal abstract class ODataWriterCore : ODataWriter, IODataOutputInStreamError
                 resourceType = this.outputContext.EdmTypeResolver.GetElementType(navigationSource);
             }
 
-            ODataUri odataUri = outputContext.MessageWriterSettings.ODataUri.Clone();
+            ODataUriSlim odataUri = new ODataUriSlim(outputContext.MessageWriterSettings.ODataUri);
 
             // Remove key for top level resource
-            if (!writingResourceSet && odataUri != null && odataUri.Path != null)
+            if (!writingResourceSet && odataUri.Path != null)
             {
                 odataUri.Path = odataUri.Path.TrimEndingKeySegment();
             }
 
             this.listener = listener;
 
-            this.scopeStack.Push(new Scope(WriterState.Start, /*item*/null, navigationSource, resourceType, /*skipWriting*/false, outputContext.MessageWriterSettings.SelectedProperties, odataUri, /*enableDelta*/ true));
+            this.scopeStack.Push(new Scope(WriterState.Start, /*item*/null, navigationSource, resourceType, /*skipWriting*/false, outputContext.MessageWriterSettings.SelectedProperties, in odataUri, /*enableDelta*/ true));

```

Comment:
Because you "touched" this line (and others that share the same pattern), I'd have suggested that you replace `/* ... */` with named parameters. However, feel free to disregard this since with the version of C# we're building the project against on ADO you'd have to apply named parameters for every parameter following the first named parameter.

Reply:
I've opted to name all parameters